### DISCUSSION TITLE
suggestion for continues recognition

### DISCUSSION THREAD

**Discussion by vovanb** [VladislavAntonyuk/MauiSamples#428] [Q&A] [answered]
Glory to Ukraine!
many thanks Vladislav for you example MauiSpeech.
I have few questions about:
1. I need continues loop with speech recognition in application, so user can say some command to app and application will confirm and will continue to listen. What is recommended  approach?
2. In your blog **https://vladislavantonyuk.github.io/articles/Speech-recognition-with-.NET-MAUI/**, you provide example to use 	`speechRecognizer.StartListening(CreateSpeechIntent(culture));` 
but in this example you use `await speechToText.ListenAsync`
Could you explain what is difference and what purpose of using Intent?
4. What is purpose of OpenAI button in sample application?

**Comment by VladislavAntonyuk** [comment, marked as answer]
Glory to heroes! Thank you for the support!

StartListening is listening until you call StopListening

ListenAsync is listening until you cancel the token.
Both methods use Continuous listening, but some platforms have silence timeout, so listening is stopped after some time. 

OpenAI here is just an example how to integrated it in app. You can start the button and say Open paint. Paint will be executed.

**Comment by vovanb** [comment]
Thanks a lot Vladislav for your response. Could you please also explain what is purpose of Intent(RecognizerIntent.ActionRecognizeSpeech)?

**Comment by VladislavAntonyuk** [comment]
In Android Intent is used to pass some parameters. the type of Intent here is Recognize speech. You also pass the culture (language) and other parameters to the SpeechRecognition.

As for the article, it describes how I created this service. It is now available in .NET MAUI CommunityToolkit, so it's better to use this package

**Comment by vovanb** [comment]
I tried to move content of Listen method to run in background :
` await Task.Run(async () =>
	{
		// Now on background thread.
		await startSpeechRecognition();
	});
`
but received error that "SpeechRecogniser should be used only from application main thread"
as I mentioned from beginning I need that application will listen all the time to user's speech and will perform some operation depends from user commands.
do you think it possible?

**Comment by VladislavAntonyuk** [comment]
The application must be in foreground, otherwise it will not work, because it can be treated as a spyware

**Reply by vovanb** [reply]
Hi Vladislav
note I tried to implement continues recognition with playing recognized text back. Note it is not working well - it always make beep before starting recognize, the played text back entering as input to recognizer.
Maybe exist another approach? For example in Android assistant "Ok Google" can be caller specific and it is always listening ...

**Reply by VladislavAntonyuk** [reply]
I didn't implement it on Android and iOS, on Windows I recorded a sound for 10 seconds and then recognized them. if a recognized text contains your command, then you start listening. It may be OK for desktop, but it takes a lot of battery resources, so may not be suitable for other devices.
